k_quants.c

```diff
@@ -2602,7 +2602,13 @@ void ggml_vec_dot_q4_K_q8_K(const int n, float * restrict s, const void * restri
 
         memcpy(utmp, x[i].scales, 12);
 
-        const uint32x2_t mins8 = {utmp[1] & kmask1, ((utmp[2] >> 4) & kmask2) | (((utmp[1] >> 6) & kmask3) << 4)};
+#ifndef _MSC_VER
+        uint32x2_t mins8 = {utmp[1] & kmask1, ((utmp[2] >> 4) & kmask2) | (((utmp[1] >> 6) & kmask3) << 4)};
+#else
+        uint32x2_t mins8;
+        mins8.n64_u32[0] = utmp[1] & kmask1;
+        mins8.n64_u32[1] = ((utmp[2] >> 4) & kmask2) | (((utmp[1] >> 6) & kmask3) << 4);
+#endif
```

Comment:
If the latter case compiles on other platforms, can we just use that everywhere?

Reply:
this is apparently per arm standard (according to vc devs): https://developercommunity.visualstudio.com/t/error-c2078-too-many-initializers-when-using-arm-n/402911 
see this analysis: https://stackoverflow.com/a/54018882 for a macro that could initialize this optimally; however, this seems like overkill for this PR.

Reply:
Apparently on at least aarch64 clang, uint32x2_t is not a structure. The standard doesn't officially support direct initialization:
> The vector data types and arrays of the vector data types cannot be initialized by direct literal
assignment. You must initialize them using one of the load intrinsics. See Table G-2 on
page G-5.

Reply:
I believe this is the portable way:
```cpp
        uint32x2_t mins8 = {0};
        mins8 = vset_lane_u32(utmp[1] & kmask1, mins8, 0);
        mins8 = vset_lane_u32(((utmp[2] >> 4) & kmask2) | (((utmp[1] >> 6) & kmask3) << 4), mins8, 1);
```
It compiles fine with aarch64 clang, but I haven't tested its correctness or whether it compiles on MSVC.

Reply:
I just tested and indeed it compiles. Performance is about the same (tested over 3 runs with openllama-7b)